waypoint_planner/src/astar_avoid/astar_avoid.cpp

```diff
@@ -156,8 +125,8 @@ void AstarAvoid::run()
   // relaying mode at startup
   state_ = AstarAvoid::STATE::RELAYING;
 
-  // start publish thread
-  publish_thread_ = std::thread(&AstarAvoid::publishWaypoints, this);
```

Comment:
So, I understand getting rid of the thread in favor of a timer, but if the timer is just going to trigger at the same rate that the node spins, why not just call `publishWaypoints` as part of the node's main loop? The timer isn't asynchronous so you aren't gaining anything except the overhead of the timer.

Reply:
Pinging @jilinzhouas since he was the original PR creator.

Reply:
Even we are running this on the real-time operating system, the timer will publish the waypoint more consistently in terms of publishing interval.